src/test/java/com/flab/bbt/product/service/ProductServiceTest.java

```diff
@@ -0,0 +1,121 @@
+package com.flab.bbt.product.service;
+
+import com.flab.bbt.exception.CustomException;
+import com.flab.bbt.product.domain.Product;
+import com.flab.bbt.product.repository.ProductRepository;
+import com.flab.bbt.product.repository.ProductRepositoryImpl;
+import com.flab.bbt.user.domain.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.doReturn;
+
+@ExtendWith(MockitoExtension.class)
+class ProductServiceTest {
+    private ProductRepository productRepository = new ProductRepositoryImpl();
+    private ProductService productService = new ProductService(productRepository);
+
+    Product product;
+    Product product2;
+
+    @BeforeEach
+    public void setUp() {
+        product = Product.builder()
+                         .name("test")
+                         .serialNum("SN00001")
+                         .imgUrl("url")
+                         .priceSale(10000)
+                         .priceDiscount(9000)
+                         .discountRate(10)
+                         .build();
+
+        product2 = Product.builder()
+                .name("test2")
+                .serialNum("SN00002")
+                .imgUrl("url")
+                .priceSale(20000)
+                .priceDiscount(18000)
+                .discountRate(10)
+                .build();
+    }
+
+    @Test
+    @DisplayName("제품 등록시 성공적으로 저장소에 저장된다.")
+    void registerSuccessTest(){
+        // given
+        // when
+        productService.register(product);
+
+        // then
+        assertThat(productRepository.findById(product.getId()).get()).isEqualTo(product);
```

Comment:
어떤 테스트를 어디에 어떻게 작성하는 것이 올바른지..!에 대해서는 저도 알아가는 중이지만 =_=.. 
요 테스트가 ProductService의 register 메서드에 대한 테스트로 적절한지 의문이 듭니당. 

```suggestion
when(productRepository.findBySerialNum("SN00001")).thenReturn(product);
when(productRepository.save(product)).thenReturn(product);
```
이런식으로 동작을 stub해주고
```
verify(productRepository, times(1)).findBySerialNum(any());
verify(productRepository, times(1)).save(any());
```
요런식으로 호출 자체가 잘 되었는지 확인해볼 수 있을 것 같습니당
(실제 저장과 조회 기능은 repository의 테스트코드에서 동작을 확인할 수 있을 것 같습니다)

Reply:
아직 stub 등에 익숙하지 않아 테스트코드가 좀 미흡하네여. 의견주신대로 수정하여 반영하였습니다. 덕분에 stub 등에 대해 좀 더 알아볼 수 있었습니다. 감사합니다.